iothub/device/src/Transport/ProtocolRoutingDelegatingHandler.cs

```diff
@@ -21,10 +21,10 @@ internal class ProtocolRoutingDelegatingHandler : DefaultDelegatingHandler
         /// <summary>
         /// After we've verified that we could open the transport for any operation, we will stop attempting others in the list.
         /// </summary>
-        private bool _transportSelectionComplete = false;
-        private int _nextTransportIndex = 0;
+        private bool _transportSelectionComplete;
+        private int _nextTransportIndex;
 
-        private SemaphoreSlim _handlerLock = new SemaphoreSlim(1, 1);
+        private readonly SemaphoreSlim _handlerLock = new SemaphoreSlim(1, 1);
```

Comment:
some with this one - does it get disposed?

Reply:
it should, but it doesn't implement dispose. I will change it.